Yosemite/Yosemite/Stores/SettingStore.swift

```diff
@@ -82,9 +82,9 @@ private extension SettingStore {
     /// Retrieves the site API information associated with the provided Site ID (if any!).
     /// This call does NOT persist returned data into the Storage layer.
     ///
-    func retrieveSiteAPI(siteID: Int64, onCompletion: @escaping (SiteAPI?, Error?) -> Void) {
-        siteAPIRemote.loadAPIInformation(for: siteID) { (siteAPI, error) in
-            onCompletion(siteAPI, error)
+    func retrieveSiteAPI(siteID: Int64, onCompletion: @escaping (Result<SiteAPI, Error>) -> Void) {
+        siteAPIRemote.loadAPIInformation(for: siteID) { result in
+            onCompletion(result)
```

Comment:
I think this can be: 

```suggestion
        siteAPIRemote.loadAPIInformation(for: siteID, completion: onCompletion)
```

🤪

Reply:
Wow, nice catch! 😄 Fixed in 4ecacfd.